src/components/utils/hooks/useMessagePreviews.tsx

```diff
@@ -67,43 +69,60 @@ const useMessagePreviews = () => {
   };
 
   useEffect(() => {
-    if (!isMessagesEnabled || !client) {
+    if (!isMessagesEnabled || !client || !currentProfile) {
       return;
     }
 
     const fetchMostRecentMessage = async (
       convo: Conversation
-    ): Promise<{ address: string; message?: Message }> => {
-      const peerAddress = convo.peerAddress.toLowerCase();
+    ): Promise<{ key: string; message?: Message }> => {
+      const key = buildConversationKey(convo.peerAddress, convo.context?.conversationId as string);
+
       const newMessages = await convo.messages({
         limit: 1,
         direction: SortDirection.SORT_DIRECTION_DESCENDING
       });
       if (newMessages.length <= 0) {
-        return { address: peerAddress };
+        return { key };
       }
-      return { address: peerAddress, message: newMessages[0] };
+      return { key, message: newMessages[0] };
     };
 
     const listConversations = async () => {
       setMessagesLoading(true);
       const newPreviewMessages = new Map(previewMessages);
       const newConversations = new Map(conversations);
-      const convos = (await client?.conversations?.list()) || [];
+      const newProfileIds = new Set(profileIds);
+      const convos = await client.conversations.list();
+      const matcherRegex = conversationMatchesProfile(currentProfile.id);
```

Comment:
Feel free to ignore, but looks we can share this between list & stream methods if we want.

Also, love this way of verifying the convo is for the correct profile while building the convo list! Super neat to see filtering working in action.

Reply:
Ya, I could make it into a hook or something. Maybe I'll do that as a followup.

Reply:
Followup would be better for this 🙏🏼